app/models/recipe_ingredient.rb

```diff
@@ -1,3 +1,8 @@
 class RecipeIngredient < ApplicationRecord
+  belongs_to :recipe
+  belongs_to :ingredient
 
+  def self.total_cost
+    joins(:ingredient).sum("ingredients.cost")
```

Comment:
I think you make this a touch harder on yourself than necessary. This method is going to return the total cost of ALL ingredients, not just ingredients in a specific recipe. A more effective approach would be to create an instance method in `Recipe` that calculates the cost of just that recipe's ingredients. Good use of AR here, it's just not quite solving the problem we're working on.

Reply:
Ahhhh I see. Thank you for the feedback. I'll continue to work on when to use instance method and which model would be the appropriate use for that specific user story.